orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/PipelineRefTriggerDeserializerSupplier.kt

```diff
@@ -24,13 +24,19 @@ import com.netflix.spinnaker.orca.pipeline.model.support.mapValue
 import com.netflix.spinnaker.orca.pipeline.model.support.listValue
 import com.netflix.spinnaker.orca.sql.pipeline.persistence.PipelineRefTrigger
 
-class PipelineRefTriggerDeserializerSupplier : CustomTriggerDeserializerSupplier {
+class PipelineRefTriggerDeserializerSupplier(
+  private val pipelineRefEnabled: Boolean
+) : CustomTriggerDeserializerSupplier {
 
   override val type: String = "pipelineRef"
 
   override val predicate: (node: JsonNode) -> Boolean
     get() = { node ->
-      node.looksLikePipeline() || node.isPipelineRefTrigger()
+      if (pipelineRefEnabled) {
+        node.looksLikePipeline() || node.isPipelineRefTrigger() //if pipelineRef enabled we deserialize PipelineTrigger as PipelineRefTrigger
```

Comment:
Not quite sure I follow this.  Don't we always want the same deserialize logic, and the flag controls what happens during serialization?

Reply:
This condition guarantees that the triggers saved under pipelineRef can still be processed even when the feature flag is turned off. Therefore, if the flag is disabled, Deck will display executions that include pipelineRef without issues

Reply:
I still don't understand why the deserialization logic needs to be different depending on pipelineRefEnabled.

Reply:
When the feature flag is enabled:

The deserializer is set up to handle both `PipelineTrigger` and `PipelineRefTrigger`.
This means that whenever Orca encounters a trigger that resembles either a `PipelineTrigger` or `PipelineRefTrigger`, it treats it as a `PipelineRefTrigger`.
Storing triggers as `PipelineRefTrigger` ensures that every execution containing a `PipelineTrigger` will store its trigger data in a format that allows the complete execution context to be retrieved later.

When the feature flag is disabled:

Orca's database may contain executions that were stored using `PipelineRefTrigger`.
Without the feature flag enabled, Orca cannot automatically deserialize a `PipelineRefTrigger`.
To maintain compatibility with older executions that used `pipelineRefTrigger` when the feature flag was enabled, the deserializer exclusively interprets `PipelineRefTrigger`.
This approach enables Orca to continue retrieving the full execution context for executions that were stored while the feature flag was active.
Additionally, new `PipelineTriggers` will no longer be stored as `PipelineRefTriggers`, ensuring consistent behavior going forward.

Therefore, in the CustomDeserializer, if the feature flag is enabled, both `PipelineTrigger` and `PipelineRefTrigger` are processed.